internal/ingress/controller/template/template.go

```diff
@@ -403,12 +403,28 @@ func buildProxyPass(host string, b interface{}, loc interface{}, dynamicConfigur
 	}
 
 	path := location.Path
-	proto := "http"
+	proto := "http://"
 
 	proxyPass := "proxy_pass"
+
+	switch location.BackendProtocol {
```

Comment:
I think this PR is the right place to also start using the annotation in existing e2e tests (GRPC is the only one coming to my mind right now).

Reply:
done